dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```diff
@@ -486,40 +460,34 @@ public CloseableClient build() {
 
             InetSocketAddress socksProxyAddress = getSocksProxyAddress(conf);
             SSLSocketFactory rawSocketFactory = conf.sslSocketFactory();
-            Supplier<Socket> simpleSocketCreator = socksProxyAddress == null
-                    ? () -> new Socket(Proxy.NO_PROXY)
-                    : () -> new Socket(new Proxy(Proxy.Type.SOCKS, socksProxyAddress));
+            // TODO(aldexis): replace this by just relying on the provided proxy set in setDefaultConfig below
+            DetachedSocketFactory simpleSocketCreator = socksProxyAddress == null
+                    ? proxy -> new Socket(proxy == null ? Proxy.NO_PROXY : proxy)
+                    : _proxy -> new Socket(new Proxy(Proxy.Type.SOCKS, socksProxyAddress));
```

Comment:
I believe this can be simplified to
```suggestion
            // TODO(aldexis): replace this by just relying on the provided proxy set in setDefaultConfig below
            DetachedSocketFactory simpleSocketCreator = proxy -> new Socket(proxy == null ? Proxy.NO_PROXY : proxy);
```
because of [this code](https://github.com/apache/httpcomponents-client/blob/f0afb8e6b73ed74c785747dea4dec43383c3d5db/httpclient5/src/main/java/org/apache/hc/client5/http/impl/io/DefaultHttpClientConnectionOperator.java#L189-L190) which creates and passes the proxy configured below in the default config with `.setSocksProxyAddress(socksProxyAddress)`. I want to make sure we have a test for this first though. For now, this ensures we stay compatible with prior logic